java/code/src/com/redhat/rhn/manager/system/entitling/SystemEntitler.java

```diff
@@ -104,6 +112,13 @@ else if (EntitlementManager.OSIMAGE_BUILD_HOST.equals(ent)) {
         server.asMinionServer().ifPresent(minion -> {
             serverGroupManager.updatePillarAfterGroupUpdateForServers(Arrays.asList(minion));
 
+            if (wasVirtEntitled && !EntitlementManager.VIRTUALIZATION.equals(ent) ||
+                    !wasVirtEntitled && EntitlementManager.VIRTUALIZATION.equals(ent)) {
```

Comment:
I wonder what happens to systems that already have the virtualization entitlement in 5.0...